testing/adios2/transform/TestBZip2Wrapper.cpp

```diff
@@ -0,0 +1,125 @@
+#include <cstdint>
+#include <iostream>
+#include <numeric> //std::iota
+#include <stdexcept>
+
+#include <adios2.h>
+
+#include <gtest/gtest.h>
+
```

Comment:
Yes, @JasonRuonanWang needs this branch that's why I started it. In fact, ZFP is basically a refactoring from his zfpman. We are checking for original data recovery in the first test. But I agree we can can make it an adios2 vs native test.